cmd/main.go

```diff
@@ -744,158 +452,448 @@ func Run() error {
+	cmds[threadPeerCmd.FullCommand()] = func () error {
 		return ThreadPeer(*threadPeerThreadID)
+	}
 
-	case threadRenameCmd.FullCommand():
+	// thread rename
+	threadRenameCmd      := threadCmd.Command("rename", "Renames a thread. Only the initiator of a thread can rename it.").Alias("mv")
+	threadRenameThreadID := threadRenameCmd.Arg("thread", "Thread ID").Required().String()
+	threadRenameName     := threadRenameCmd.Arg("name", "The name to rename the thread to").Required().String()
+	cmds[threadRenameCmd.FullCommand()] = func () error {
 		return ThreadRename(*threadRenameName, *threadRenameThreadID)
+	}
 
-	case threadUnsubscribeCmd.FullCommand():
-		return ThreadUnsubscribe(*threadUnsubscribeThreadID)
+	// thread abandon
+	threadAbandonCmd      := threadCmd.Command("abandon", "Abandon a thread. If no one is else remains participating, the thread dissipates.").Alias("unsubscribe").Alias("leave").Alias("remove").Alias("rm")
+	threadAbandonThreadID := threadAbandonCmd.Arg("thread", "Thread ID").Required().String()
+	cmds[threadAbandonCmd.FullCommand()] = func () error {
+		return ThreadAbandon(*threadAbandonThreadID)
+	}
+
+	// thread snapshot
+	// @todo are snapshots blocks?
+	threadSnapshotCmd := threadCmd.Command("snapshot", "Manage thread snapshots").Alias("snapshots")
 
-	case threadSnapshotCreateCmd.FullCommand():
+	// thread snapshot create
+	threadSnapshotCreateCmd := threadSnapshotCmd.Command("create", "Snapshots all threads and pushes to registered cafes").Alias("make")
+	cmds[threadSnapshotCreateCmd.FullCommand()] = func () error {
 		return ThreadSnapshotCreate()
+	}
 
-	case threadSnapshotSearchCmd.FullCommand():
+	// thread snapshot search
+	threadSnapshotSearchCmd  := threadSnapshotCmd.Command("search", "Searches the network for thread snapshots").Alias("find").Default()
+	threadSnapshotSearchWait := threadSnapshotSearchCmd.Flag("wait", "Stops searching after [wait] seconds have elapse (max 30s)").Short('w').Default("2").Int()
+	cmds[threadSnapshotSearchCmd.FullCommand()] = func () error {
 		return ThreadSnapshotSearch(*threadSnapshotSearchWait)
+	}
 
-	case threadSnapshotApplyCmd.FullCommand():
+	// thread snapshot apply
+	threadSnapshotApplyCmd  := threadSnapshotCmd.Command("apply", "Applies a single thread snapshot")
+	threadSnapshotApplyID   := threadSnapshotApplyCmd.Arg("snapshot", "The ID of the snapshot to apply").Required().String()
+	threadSnapshotApplyWait := threadSnapshotApplyCmd.Flag("wait", "Stops searching after [wait] seconds have elapse (max 30s)").Short('w').Default("2").Int()
+	cmds[threadSnapshotApplyCmd.FullCommand()] = func () error {
 		return ThreadSnapshotApply(*threadSnapshotApplyID, *threadSnapshotApplyWait)
+	}
+
+	// thread file
+	threadFileCmd, threadFileCallback := threadFilesCommand(threadCmd, []string{"files", "file"})
+	cmds[threadFileCmd.FullCommand()] = threadFileCallback
+
+	// ================================
+	// @todo are tokens blocks?
```

Comment:
Nope, tokens are local-only - essentially passwords to get a session token (JWT) to a cafe.